pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -942,12 +942,20 @@ def contains_merged_edge(self, edge: MergedEdge) -> bool:
             return edge == other
         return False
 
-    def create_merged_edge(self, source_id: NodeIdLike, target_id: NodeIdLike, edges: Iterable[Edge]) -> MergedEdge:
+    def create_merged_edge(
+        self, source_id: NodeIdLike, target_id: NodeIdLike, edges: Iterable[Edge | MergedEdge]
+    ) -> MergedEdge:
         if len(list(edges)) == 0:
             raise ValueError(f'Cannot build KCFG MergedEdge with no edges: {edges}')
         source = self.node(source_id)
         target = self.node(target_id)
-        merged_edge = KCFG.MergedEdge(source, target, tuple(edges))
+        flatten_edges: list[KCFG.Edge] = []
+        for edge in edges:
+            if isinstance(edge, KCFG.MergedEdge):
+                flatten_edges.extend(edge.edges)
+            else:
+                flatten_edges.append(edge)
+        merged_edge = KCFG.MergedEdge(source, target, tuple(flatten_edges))
         self.add_successor(merged_edge)
```

Comment:
Should we be checking that each `edge` in `flatten_edges` has the correct `source` and `target`? We may want to either:

- Make sure by updating their `source` and `target`, so we don't get serialization issues, or
- Assert that they already have the correct `source` and `target`.

Reply:
I'm not sure about the correct `source` and `target` of `flatten_edges`. We deleted them before `create_merged_edge`. I guess $edge.source \implies merged.source$ and $edge.target \implies merged.target$. But this validation needs `CTermSymbolic` or `KCFGExplore` to return a boolean result. KCFG is just a syntactic structure. How about introducing a `KCFGChecker` to check the semantics correctness of KCFG?